### PR TITLE
Debugging `zenml clean`

### PR DESCRIPTION
**Pull request by bcdurak**
## Describe changes
I fixed the `zenml clean` CLI command so that it works when current zenml version < DB version.

## Pre-requisites
Please ensure you have done the following:
- [X] I have read the **CONTRIBUTING.md** document.
- [X] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] If I have added an integration, I have updated the [integrations](https://docs.zenml.io/stacks-and-components/component-guide) table and the [corresponding website section](https://zenml.io/integrations).
- [ ] I have added tests to cover my changes.
- [X] I have based my new branch on `develop` and the open PR is targeting `develop`. If your branch wasn't based on develop read [Contribution guide on rebasing branch to develop](https://github.com/zenml-io/zenml/blob/main/CONTRIBUTING.md#-pull-requests-rebase-your-branch-on-develop).

## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

### CODE REVIEW

File: src/zenml/cli/server.py [hunk 1]
```diff
@@ -251,8 +251,8 @@ def down() -> None:
         deployer.remove_server(server.config.name)
         cli_utils.declare("The local ZenML dashboard has been shut down.")
 
-    gc = GlobalConfiguration()
-    gc.set_default_store()
+        gc = GlobalConfiguration()
```
Comment (stefannica) [inline src/zenml/cli/server.py:254]: good catch !

File: src/zenml/cli/base.py [hunk 2]
```diff
@@ -318,12 +324,22 @@ def clean(ctx: click.Context, yes: bool = False, local: bool = False) -> None:
...
+            config_version = version.parse(global_version)
+
+            if config_version > curr_version:
+                error(
+                    "Due to this version mismatch, ZenML can not detect and "
+                    "shut down any running dashboards or clean any resources "
+                    "related to the active stack."
+                )
         _delete_local_files(force_delete=yes)
-        return
```
Comment (stefannica) [inline src/zenml/cli/base.py:326]: If I remember correctly, the `local` clean is only supposed to clean the local artifacts stored for the active stack and nothing else. You definitely need to keep this `return`.

File: src/zenml/cli/base.py [hunk 3]
```diff
@@ -318,12 +324,22 @@ def clean(ctx: click.Context, yes: bool = False, local: bool = False) -> None:
...
+        global_version = GlobalConfiguration().version
+        if global_version is not None:
+            config_version = version.parse(global_version)
+
+            if config_version > curr_version:
+                error(
+                    "Due to this version mismatch, ZenML can not detect and "
+                    "shut down any running dashboards or clean any resources "
+                    "related to the active stack."
+                )
```
Comment (stefannica) [inline src/zenml/cli/base.py:341]: but you're already shutting down the running dashboard outside of this check, at line 327.

**Review by stefannica — CHANGES_REQUESTED** [review]

File: src/zenml/cli/server.py (hunk 1 above)
Comment (bcdurak) [reply src/zenml/cli/server.py:254]: thanks :)

**Review by stefannica — APPROVED** [review]
splendid job !

**Pull request merged**